src/modules/adhs/src/lib/Adhs.c

```diff
@@ -0,0 +1,360 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <errno.h>
+#include <stdatomic.h>
+#include <version.h>
+#include <CommonUtils.h>
+#include <Logging.h>
+#include <Mmi.h>
+#include <regex.h>
+
+#include "Adhs.h"
+
+static const char* g_adhsModuleInfo = "{\"Name\": \"Adhs\","
+    "\"Description\": \"Provides functionality to observe and configure Azure Device Health Services (ADHS)\","
+    "\"Manufacturer\": \"Microsoft\","
+    "\"VersionMajor\": 1,"
+    "\"VersionMinor\": 0,"
+    "\"VersionInfo\": \"Copper\","
+    "\"Components\": [\"Adhs\"],"
+    "\"Lifetime\": 2,"
+    "\"UserAccount\": 0}";
+
+static const char* g_adhsModuleName = "Adhs module";
+static const char* g_adhsComponentName = "Adhs";
+
+static const char* g_reportedOptInObjectName = "optIn";
+static const char* g_desiredOptInObjectName = "desiredOptIn";
+
+static const char* g_permissionConfigPattern = "\\bPermission\\s*=\\s*([\\\"'])([A-Za-z0-9]*)\\1";
+static const char* g_permissionConfigName = "Permission";
+static const char* g_permissionConfigMapKeys[] = { "None", "Required", "Optional" };

```

Comment:
>{ "None", "Required", "Optional" }; [](http://example.com/codeflow?start=49&length=35)

nit: here and elsewhere, no spaces after { and before } for initializations

Reply:
Thanks, fixed!